lib/Date/Utility.pm

```diff
@@ -840,6 +838,35 @@ sub months_ahead {
     return month_number_to_abbrev($new_month) . '-' . $new_year;
 }
 
+=head2 move_to_nth_dow
+
+Takes an integer as an ordinal and a day of week representation
+
+The following are all equivalent:
+C<move_to_nth_dow(3, 'Monday')>
+C<move_to_nth_dow(3, 'Mon')>
+C<move_to_nth_dow(3, 1)>
+
+Returning the 3rd Monday of the month represented by the object or
+C<undef> if it does not exist.
+
+An exception is thrown on improper day of week representations.
+
+=cut
+
+sub move_to_nth_dow {
+    my ($self, $nth, $dow_abb) = @_;
+
+    $dow_abb //= 'undef';    # For nicer error reporting below.
+
+    my $dow = $days_to_num{lc $dow_abb} // croak 'Invalid day of week. We got [' . $dow_abb . ']';
+
+    my $dow_first = ($self->day_of_month - $self->day_of_week) % 7;
+    my $dom = ($dow + 7 - $dow_first) % 7 + ($nth - 1) * 7 + 1;
+
+    return try { Date::Utility->new(join '-', $self->year, $self->month, $dom) };
+}
```

Comment:
I did extensive testing (not committed in favor of fast running tests) and it was producing correct results.  I am not sure what you're demonstrating here.